### PR TITLE
TCES-54: Employer Add Form

### PR DESCRIPTION
**Pull request by thardchi**
https://github.com/uoftblueprint/tces/issues/54
# Description

Component lets us add employers. Screenshot below:

Fixes # (54)

## Sample curl
Ignore if PR does not affect an endpoint.
## Response
Ignore if PR does not affect an endpoint.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

Screen Shot 2023-01-05 at 11 21 57 PM

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useEffect, useState } from "react";
 import logo from "./logo.svg";
 import "./App.css";
 import Communication from "./components/Communication";
+import EpmployerAddForm from "./components/EmployerAddForm";
```
Comment (helen-li) [inline client/src/App.tsx:5]:
```suggestion
import EmployerAddForm from "./components/EmployerAddForm";
```
Just a minor typo in this file and in the components file

File: client/src/components/EmployerAddForm.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    <Box style={{borderStyle:"dashed", borderWidth:"0 0 1.8px 0"}} borderColor="primary.main" width="100%" paddingTop={1} paddingLeft={1.5}>
+                    <Typography gutterBottom variant="h6" align="left">Company Information</Typography>
+                    </Box>
+                    <CardContent>
+    
+                    <form>
+                        <Grid container spacing={1}>
+                            <Grid xs={12} item>
+                            <TextField label="Business Name" placeholder="Enter business name" variant="outlined" fullWidth/>
+                            {/* <Typography margin={0.5} fontSize={11}align="left" color="text.secondary">*Required</Typography> */}
```
Comment (helen-li) [inline client/src/components/EmployerAddForm.tsx:24]: Would recommend removing commented out code if no longer needed to reduce clutter heh

File: client/src/components/EmployerAddForm.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                            <Grid xs={12} sm={6} item>
+                            <TextField type="email" label="General Email" placeholder="Enter general email" variant="outlined" fullWidth/>
+                            <FormHelperText>*Required</FormHelperText>
+                            </Grid>
+    
+                            <Grid xs={12} sm={6} item>
+                            <TextField type="url" label="Website" placeholder="Enter website" variant="outlined" fullWidth/>
+                            </Grid>
+    
+                            <Grid xs={12} item>
```
Comment (helen-li) [inline client/src/components/EmployerAddForm.tsx:50]: Might be useful to extract these Grid components and write them as a reusable component for you to pass the different labels and placeholders in to reduce repetition & we probably need similar grids for other forms in the future - feel free to take a quick look at #17 `FormTextField.tsx` for an example of what I mean by reusable component & we can chat more about it too

File: client/src/components/EmployerAddForm.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                            <Grid xs={12} sm={6} item>
+                            <TextField label="Postal Code" placeholder="Enter postal code" variant="outlined" fullWidth/>
+                            <FormHelperText>*Required</FormHelperText>
+                            </Grid>
+                        </Grid>
+                    </form>
+                    </CardContent>
+                  </Card>
+    
+                {/* Add Seconday Address Button */}
```
Comment (helen-li) [inline client/src/components/EmployerAddForm.tsx:91]: Nice comments!

**Review by helen-li — CHANGES_REQUESTED** [review]
Good work!!! Dropped a few comments and suggestions throughout, also I noticed there were multiple layouts in the Employer Add page on our Figma - just wondering if it's within the scope of this ticket to write them all?

**Review by jordanjanakievski — COMMENTED** [review]
Everything worked great and the form looks terrific. I cannot explicitly remember if you were worried about the Province dropdown, but everything looking good on my end as they were left-aligned.

Helen has covered everything that I noticed. Once you make those changes, you can ping me again and I will approve.

**Review by helen-li — APPROVED** [review]
Niceee super good work Thardchi!!!

**Review by jordanjanakievski — APPROVED** [review]
The changes are good

**Pull request merged**